auto-value-gson/build.gradle

```diff
@@ -3,14 +3,15 @@ import org.gradle.internal.jvm.Jvm
 apply plugin: 'java-library'
 apply plugin: 'maven-publish'
 
-sourceCompatibility = JavaVersion.VERSION_1_7
-targetCompatibility = JavaVersion.VERSION_1_7
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
 
 dependencies {
     api project(':auto-value-gson-annotations')
     api deps.gson
     implementation 'com.squareup:javapoet:1.9.0'
-    api 'com.google.auto.value:auto-value:1.4.1'
+    implementation 'com.google.auto.value:auto-value:1.6'
+    implementation 'com.google.auto.value:auto-value-annotations:1.6'
```

Comment:
I think it should be:
```
api 'com.google.auto.value:auto-value:1.6'
compileOnly 'com.google.auto.value:auto-value-annotations:1.6'
```

Reply:
Description for ```api```:
> When a module includes an api dependency, it's letting Gradle know that the module wants to transitively export that dependency to other modules, so that it's available to them at both runtime and compile time.

But it seems that compilation classpath and annotation processing classpath are separated from each other now, so there is no point for annotation processors to provide their dependencies transitively.

 Now about ```compileOnly``` dependency. Description says:
> Gradle adds the dependency to the compilation classpath only (it is not added to the build output). This is useful when you're creating an Android library module and you need the dependency during compilation, but it's optional to have present at runtime.

With this I still have compilation error mentioned earlier. Maybe this happens because for annotation processors compilation and runtime is same thing, I'm actually don't fully understand this new annotation processors handling.